server/datastore/mysql/sessions.go

```diff
@@ -11,8 +11,10 @@ import (
 
 func (ds *Datastore) SessionByKey(ctx context.Context, key string) (*fleet.Session, error) {
 	sqlStatement := `
-		SELECT * FROM sessions
-			WHERE ` + "`key`" + ` = ? LIMIT 1
+		SELECT s.*, u.api_only FROM sessions s
+		LEFT JOIN users u
```

Comment:
Curious: Can we have user-less sessions? (An entry on `sessions` which does not have the corresponding user in `users`)

Reply:
Technically, yes - there's no FK constraint on the user_id column in sessions, so that's just to be extra-safe when touching session-related code (given that the old implementation did not join with users, I didn't want to add a new constraint here that the user had to exist). You'll notice that for ListSessionsByUser, I *did* use an inner join, as it doesn't make sense to list sessions for a user that does not exist.